compiler/x/codegen/ControlFlowEvaluator.cpp

```diff
@@ -2239,7 +2239,7 @@ TR::Register *OMR::X86::TreeEvaluator::sucmpeqEvaluator(TR::Node *node, TR::Code
    if (cg->enableRegisterInterferences())
       cg->getLiveRegisters(TR_GPR)->setByteRegisterAssociation(targetRegister);
 
-   generateRegInstruction(SETNE1Reg, node, targetRegister, cg);
+   generateRegInstruction(node->getOpCodeValue() == TR::sucmpeq ? SETE1Reg : SETNE1Reg, node, targetRegister, cg);
```

Comment:
`sucmpeq` is deprecated as it's functionally identical to `scmpeq` (see #2657). Is there a reason for using it here?

Reply:
Nevermind. Looking at the change again, my question doesn't make sense.

Reply:
`sucmpeq` or `sucmpne` is converted from `ifsucmpeq` or `ifsucmpne` by CFG Simplfier through some opcode APIs. `ifsucmpeq` and `ifsucmpne` are generated by tree simplification. @kobinau Can we get a PR to clean the use of them in https://github.com/eclipse/omr/blob/master/compiler/optimizer/OMRSimplifierHandlers.cpp?

Reply:
In OMRSimplifierHandlers.cpp, both `ifsucmpeq` and `ifsucmpne` only occur as parameters to functions `intCompareNarrower` and `longCompareNarrower` both of which specifically want those IL opcode accompanied with `ifscmpeq.`
Should I be modifying those functions as well?

Reply:
Until the removal is complete fixing the evaluator makes sense to prevent bugs - the removal is good as well so we can merge that once this is sorted.